Docker/itk-base/Dockerfile

```diff
@@ -16,7 +17,11 @@ RUN curl -O https://cmake.org/files/v3.4/cmake-3.4.3.tar.gz && \
     cd /tmp/cmake-build && \
     ../cmake-3.4.3/bootstrap && \
     make -j$(nproc) && \
-    ./bin/cmake -DCMAKE_BUILD_TYPE:STRING=Release . && \
+    cd .. && \
+    mkdir /tmp/cmake-ssl-build && \
+    cd /tmp/cmake-ssl-build && \
+    ../cmake-build/bin/cmake -DCMAKE_USE_OPENSSL:BOOL=ON \
+    -DCMAKE_BUILD_TYPE:STRING=Release ../cmake-3.4.3 && \
```

Comment:
You can actually avoid two separate builds and simplify the process by piggy-backing off the bootstrap script.  The CMake build process works by using the bootstrap script to manually build a mini-cmake (it ends up in `./Bootstrap.cmk/cmake`), which is then used to run the CMake configure step for the full build.  The script also gives you the ability to pass additional options that get forwarded on to the cmake used in said configure step:
```
cd /tmp/cmake-build && \
../cmake-3.4.3/bootstrap --parallel=$(nproc) -- \
  -DCMAKE_BUILD_TYPE:STRING=Release \
  -DCMAKE_USE_OPENSSL:BOOL=ON &&\
make -j$(nproc) && \
make install && \
```